dart/lib/src/sentry_client.dart

```diff
@@ -131,18 +132,42 @@ class SentryClient {
       return event;
     }
 
-    if (event.exceptions?.isNotEmpty ?? false) return event;
+    if (event.exceptions?.isNotEmpty ?? false) {
+      return event;
+    }
+
+    final isolateName = getIsolateName();
```

Comment:
The problem here is that `Isolate.current` returns the current Isolate, but if the error was thrown by another Isolate and bubbled up on this one, it'd use the wrong Isolate data.
I feel that the event itself (the integration that created the exception) should already have set this info (isolateName) in the event, so we can reuse to create the thread metadata.

Reply:
How should that work then for `Sentry.captureException()`? There's no integration, which could add it.

Reply:
One solution I can think of is that integrations supply the isolate info as a hint and then do a `hint['isolate'] ?? Isolate.current.debugName` to figure out the correct isolate.

Reply:
`Sentry.captureException` and `captureMessage` if called manually would use the current Isolate.
Integrations usually call `Sentry.captureEvent` and in this case, we could set the Isolate data to the event.
Using `hints` is another option, but we'd need to move to a `Map` first.

Reply:
Then I would say we keep it like it is, because it most likely covers most use cases?